geomesa-memory/geomesa-cqengine-datastore/src/main/scala/org/locationtech/geomesa/memory/cqengine/datastore/GeoCQEngineFeatureStore.scala

```diff
@@ -57,17 +39,62 @@ class GeoCQEngineFeatureStore(engine: GeoCQEngine, entry: ContentEntry, query: Q
   override def getCountInternal(query: Query): Int =
     SelfClosingIterator(engine.query(query.getFilter)).length
 
-  override def getReaderInternal(query: Query): FeatureReader[SimpleFeatureType, SimpleFeature] = {
-    import scala.collection.JavaConversions._
-    new DelegateFeatureReader(engine.sft, new DelegateFeatureIterator(engine.query(query.getFilter)))
-  }
+  override def getReaderInternal(query: Query): FeatureReader[SimpleFeatureType, SimpleFeature] =
+    new DelegateFeatureReader(engine.sft, new DelegateFeatureIterator(engine.query(query.getFilter).asJava))
+
+  override def getQueryCapabilities: QueryCapabilities = GeoCQEngineQueryCapabilities
+
+  override def canEvent: Boolean = true
```

Comment:
This just makes me wonder if the right 'canXYZ' calls are set for this FeatureStore implementation....  hmm....  should canFilter be true?

Reply:
ah, probably? does cqengine handle the full filter, or just the parts it can extract and index?

Reply:
Can't remember.  Maybe we should enter a ticket to sort through that?